resources/aws/environment.go

```diff
@@ -96,7 +98,8 @@ func NewEnvironment(ctx *pulumi.Context, options ...func(*Environment)) (Environ
 	env.envDefault = getEnvironmentDefault(config.FindEnvironmentName(env.InfraEnvironmentNames(), awsConfigNamespace))
 
 	awsProvider, err := sdkaws.NewProvider(ctx, string(config.ProviderAWS), &sdkaws.ProviderArgs{
-		Region: pulumi.String(env.Region()),
+		Region:  pulumi.String(env.Region()),
+		Profile: pulumi.String(env.Profile()),
```

Comment:
Note, perhaps for later: It may be useful to detect command is already run with credentials and `AWS_PROFILE=<empty>` (probably something like `AWS_SECRET_ACCESS_KEY/AWS_SESSION_TOKEN` being set, but need to be verified further) to avoid breaking already authed context.

Reply:
Yes, that's what I have with an active session: 

```
AWS_VAULT=sso-agent-sandbox-account-admin
AWS_REGION=us-east-1
AWS_DEFAULT_REGION=us-east-1
AWS_ACCESS_KEY_ID=something
AWS_SECRET_ACCESS_KEY=something
AWS_SESSION_TOKEN=something
AWS_CREDENTIAL_EXPIRATION=2024-07-01T14:57:36Z
```